### DISCUSSION TITLE
Why isn't obj.primary_ipv4 returning anything?

### DISCUSSION THREAD

**Discussion by WhaleJ84** [netbox-community/netbox#6031] [Q&A] [answered]
Hey all,

I've got a test instance of Netbox going for developing purposes and I'm currently just trying to add a 'custom link' to the devices added in Netbox.
The goal is that: *if device has IPv4 address, show this button* and yet I'm having issues.

The custom field currently has the text field:
```
{% if obj.primary_ipv4 != None %}Add to monitor{% endif %}
```
And link field:
```
http://monitor.example.com/?name={{ obj.name }}&ipv4={{ obj.primary_ipv4 }}/
```

I have a device named `test_device` with primary IPv4/6 set to `192.168.1.1/24` and `fd00::c0a8:101/120` respectively.
I set the IPv6 just on the off-chance that the code was bugged and it was trying to return a v6 address instead.

As per the if statement, the button appears at the top of the page and yet the link directs to `http://monitor.example.com/?name=test_device&ipv4=/` without the IPv4 address included.

Am I doing something wrong or is this a bug?

**Comment by ktims** [comment, marked as answer]
The `primary_ip4` (note, not `primary_ipv4`) is an `IPAddress` object. You need `{{ obj.primary_ip4.address.ip }}` to get the bare address (sans netmask).

**Reply by WhaleJ84** [reply]
When I change the link field to use that logic, the button gets greyed out with a warning symbol.
The help-tip for the button says: `'dcim.models.devices.Device object' has no attribute 'primary_ipv4'`.

Exact link snippet for reference:
```
http://monitor.example.com/?name={{ obj.name }}&ipv4={{ obj.primary_ipv4.address.ip }}/
```

**Reply by candlerb** [reply]
As @ktims said, the attribute is `primary_ip4` not `primary_ipv4`

**Reply by WhaleJ84** [reply]
Ah, that's it! Devil was in the details. I read it over a few times and still somehow missed the **v** is omitted. Thanks!
